src/main/java/crud/web/filters/AuthenticationFilter.java

```diff
@@ -0,0 +1,46 @@
+package crud.web.filters;
+
+import crud.lib.Injector;
+import crud.service.abstraction.DriverService;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector INJECTOR = Injector.getInstance("crud");
+    private final DriverService driverService
+            = (DriverService) INJECTOR.getInstance(DriverService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String servletPath = req.getServletPath();
+        if (servletPath.equals("/login") || servletPath.equals("/drivers/create")) {
+            chain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
```

Comment:
Please, check the 5th point in [common mistakes](https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/09_filters_checklist.html)